server/chat-plugins/chatlog.ts

```diff
@@ -356,24 +551,48 @@ export const pages: PageTable = {
 		}
 
 		void accessLog.writeLine(`${user.id}: <${roomid}> ${date}`);
-
 		this.title = '[Logs] ' + roomid;
+		let limit = null;
+		let search;
+		if (opts?.startsWith('search-')) {
+			let [input, limitString] = opts.split('--limit-');
+			input = input.slice(7);
+			search = Dashycode.decode(input);
+			if (limitString) {
+				limit = parseInt(limitString);
+			} else {
+				limit = null;
```

Comment:
Wait a minute, I don't see handling for the `--limit-all` case.

```suggestion
		/** null = no limit */
		let limit: number | null = null;
		let search;
		if (opts?.startsWith('search-')) {
			let [input, limitString] = opts.split('--limit-');
			input = input.slice(7);
			search = Dashycode.decode(input);
			if (limitString) {
				limit = parseInt(limitString) || null;
			} else {
				limit = 500;
```